pkg/doc/bbs/bbs12381g2pub/bbs.go

```diff
@@ -83,7 +88,61 @@ func (b BBSG2Pub) Verify(messages [][]byte, sigBytes, pubKeyBytes []byte) error
 	return errors.New("BLS12-381: invalid signature")
 }
 
-func getB(s *bls.FR, messages []*SignatureMessage, key *PublicKey) (*bls.G1Affine, error) {
+// Sign signs the one or more messages using private key in compressed form.
+func (bbs *BBSG2Pub) Sign(messages [][]byte, privKeyBytes []byte) ([]byte, error) {
+	privKey, err := UnmarshalPrivateKey(privKeyBytes)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal private key: %w", err)
+	}
+
+	if len(messages) == 0 {
+		return nil, errors.New("messages are not defined")
+	}
+
+	return bbs.SignWithKeyPair(messages, privKey, privKey.PublicKey())
+}
+
+// SignWithKeyPair signs the one or more messages using BBS+ key pair.
+func (bbs *BBSG2Pub) SignWithKeyPair(messages [][]byte, privKey *PrivateKey, pubKey *PublicKey) ([]byte, error) {
+	var err error
+
+	messagesFr := make([]*SignatureMessage, len(messages))
+	for i := range messages {
+		messagesFr[i], err = ParseSignatureMessage(messages[i])
+		if err != nil {
+			return nil, fmt.Errorf("parse signature message: %w", err)
```

Comment:
add the index `i` in the error message to help troubleshoot errors

Reply:
Improved.